core/sdk-core/src/main/java/software/amazon/awssdk/core/async/AsyncResponseTransformer.java

```diff
@@ -116,11 +117,27 @@
      * @param path        Path to file to write to.
      * @param <ResponseT> Pojo Response type.
      * @return AsyncResponseTransformer instance.
+     * @see #toFile(Path, FileTransformerConfiguration)
      */
     static <ResponseT> AsyncResponseTransformer<ResponseT, ResponseT> toFile(Path path) {
         return new FileAsyncResponseTransformer<>(path);
     }
 
+    /**
+     * Creates an {@link AsyncResponseTransformer} that writes all the content to the given file with the specified configuration.
+     * By default, if the file already exists, an exception will be thrown, and in the event of an error, the SDK will attempt to
+     * delete the file. This behavior can be configured via {@link FileTransformerConfiguration}.
+     *
+     * @param path        Path to file to write to.
+     * @param config      configuration for the transformer
+     * @param <ResponseT> Pojo Response type.
+     * @return AsyncResponseTransformer instance.
+     * @see FileTransformerConfiguration
+     */
+    static <ResponseT> AsyncResponseTransformer<ResponseT, ResponseT> toFile(Path path, FileTransformerConfiguration config) {
+        return new FileAsyncResponseTransformer<>(path, config);
+    }
```

Comment:
You probably considered it already, but a consumer-builder would be nice.

Reply:
Yeah, thought about it and was concerned that having too many overloads may make it harder to use; I guess consistency takes precedence.